File: lib/data/models/branch_model.dart
import '../../domain/entities/branch.dart';
import '../database/app_database.dart';

/// 分支数据模型 - Entity ↔ Data 转换
extension BranchDataExtension on BranchData {
  /// 转换为领域实体
  Branch toEntity() {
    return Branch(
      id: id,
      repositoryId: repositoryId,
      name: name,
      parentBranchId: parentBranchId,
      isMain: isMain,
      color: color ?? '#3B82F6',
      isDeleted: isDeleted,
      createdAt: createdAt,
      updatedAt: updatedAt,
    );
  }
}

/// 分支实体扩展 - 转换为 drift 数据
extension BranchEntityExtension on Branch {
  /// 转换为 drift 数据对象
  BranchData toData() {
    return BranchData(
      id: id,
      repositoryId: repositoryId,
      name: name,
      parentBranchId: parentBranchId,
      isMain: isMain,
      color: color,
      isDeleted: isDeleted,
      createdAt: createdAt,
      updatedAt: updatedAt,
    );
  }
}

File: web/src/data/models/branch-model.ts
import { Branch } from '../../domain/entities/branch';

export interface BranchRecord {
  id: string;
  repositoryId: string;
  name: string;
  parentBranchId: string | null;
  isMain: number;
  color: string;
  isDeleted: number;
  createdAt: string;
  updatedAt: string;
}

export const BranchModel = {
  toRecord(entity: Branch): BranchRecord {
    return {
      id: entity.id,
      repositoryId: entity.repositoryId,
      name: entity.name,
      parentBranchId: entity.parentBranchId,
      isMain: entity.isMain ? 1 : 0,
      color: entity.color,
      isDeleted: entity.isDeleted ? 1 : 0,
      createdAt: entity.createdAt.toISOString(),
      updatedAt: entity.updatedAt.toISOString(),
    };
  },

  toEntity(record: BranchRecord): Branch {
    return new Branch({
      id: record.id,
      repositoryId: record.repositoryId,
      name: record.name,
      parentBranchId: record.parentBranchId,
      isMain: record.isMain === 1,
      color: record.color,
      isDeleted: record.isDeleted === 1,
      createdAt: new Date(record.createdAt),
      updatedAt: new Date(record.updatedAt),
    });
  },
};
